operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java

```diff
@@ -904,6 +981,67 @@ public NextAction apply(Packet packet) {
         return doNext(packet);
       }
     }
+
+    private boolean shouldNotRestartAfterOnlineUpdate(String dynamicUpdateResult) {
+      boolean result = false;
+      if (ProcessingConstants.MII_DYNAMIC_UPDATE_SUCCESS.equals(dynamicUpdateResult)) {
+        result = true;
+      } else if (ProcessingConstants.MII_DYNAMIC_UPDATE_RESTART_REQUIRED.equals(dynamicUpdateResult)) {
+        result = MIINonDynamicChangesMethod.CommitUpdateOnly.equals(
+            Optional.ofNullable(info)
+             .map(DomainPresenceInfo::getDomain)
+             .map(Domain::getSpec)
+             .map(DomainSpec::getConfiguration)
+             .map(Configuration::getModel)
+             .map(Model::getOnlineUpdate)
+             .map(OnlineUpdate::getOnNonDynamicChanges)
+             .orElse(MIINonDynamicChangesMethod.CommitUpdateOnly));
+      }
+      return result;
+    }
+
+    private void setOnlineUpdateNeedRestartCondition(Packet packet) {
+      String dynamicUpdateRollBackFile = Optional.ofNullable((String)packet.get(
+          ProcessingConstants.MII_DYNAMIC_UPDATE_WDTROLLBACKFILE))
+          .orElse("");
+      updateDomainConditions("Online update completed successfully, but the changes require "
+              + "restart and the domain resource specified "
+              + "'spec.configuration.model.onlineUpdate.onNonDynamicChanges=CommitUpdateOnly' or not set."
+              + " The changes are committed but the domain require manually restart to "
+              + " make the changes effective. The changes are: "
```

Comment:
We should meet to word-smith this. Also, ideally such text would be maintained in a log messages file instead of existing directly in the java. Once we're done, Rosemary might want to chime in.